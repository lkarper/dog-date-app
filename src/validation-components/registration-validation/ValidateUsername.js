import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

const ValidateUsername = (props) => {
    const { usernameValidationError, usernameExists } = props;

    useEffect(() => {
        const { username, setUsernameValidationError } = props;
        if (username.trim().length >= 3 && username.trim().length <= 72) {
            setUsernameValidationError(null);
        } else {
            setUsernameValidationError('Invalid username.');
        }
        
    }, [props]);

    const accountExistsText = (<>The username {usernameExists} is already linked to a registered account.  If that account belongs to you, {<Link to='/login'>click here</Link>} to log in.</>);

    if (usernameValidationError) {
        return (
            <p id='username-validation'>Username required and must be 3-72 characters in length.
                <br />
                {usernameExists && accountExistsText}
            </p>
        );
    }

    return (
        <p id='username-validation'>Username properly formatted.
            <br />
            {usernameExists && accountExistsText}
        </p>
    );
}

export default ValidateUsername;