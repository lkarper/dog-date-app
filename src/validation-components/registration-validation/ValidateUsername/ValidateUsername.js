import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const ValidateUsername = (props) => {
    const { 
        username,
        usernameValidationError, 
        usernameExists,
        setUsernameValidationError 
    } = props;

    useEffect(() => {
        if (username.trim().length >= 3 && username.trim().length <= 72) {
            setUsernameValidationError('');
        } else {
            setUsernameValidationError('Username required and must be 3-72 characters in length.');
        }
        
    }, [username, setUsernameValidationError]);

    const accountExistsText = (
        <>
            <span
                style={{ color: 'red' }}
            >
                The username {usernameExists} is already linked to a registered account.
            </span>
            <br />
            <span
                style={{ color: 'black' }}
            >
                    If that account belongs to you,{' '} 
                    <span 
                        style={{ color: 'hsl(245, 100%, 50%)' }}
                    >
                        {<Link to='/login'>click here</Link>}
                    </span> 
                    {' '}to login.            
            </span>
        </>
    );

    if (usernameValidationError) {
        return (
            <p
                className='ValidateUserName__validator error' 
                id='username-validation'
            >
                {usernameValidationError}
                <br />
                {usernameExists && accountExistsText}
            </p>
        );
    }

    return (
        <p 
            className='ValidateUserName__validator valid'
            id='username-validation'
        >
            Username properly formatted.
            <br />
            {usernameExists && accountExistsText}
        </p>
    );
}

ValidateUsername.defaultProps = {
    username: '',
    usernameValidationError: 'Username required and must be 3-72 characters in length.', 
    usernameExists: '',
    setUsernameValidationError: () => {},
};

ValidateUsername.propTypes = {
    username: PropTypes.string.isRequired,
    usernameValidationError: PropTypes.string.isRequired,
    usernameExists: PropTypes.string.isRequired,
    setUsernameValidationError: PropTypes.func.isRequired,
};

export default ValidateUsername;
