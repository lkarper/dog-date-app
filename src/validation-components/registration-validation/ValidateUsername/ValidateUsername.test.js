import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateUsername from './ValidateUsername';

describe('ValidateUsername component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateUsername />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateUsername />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
