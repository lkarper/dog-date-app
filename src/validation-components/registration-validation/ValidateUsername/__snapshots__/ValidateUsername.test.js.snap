// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateUsername component renders the UI as expected 1`] = `
<ValidateUsername
  setUsernameValidationError={[Function]}
  username=""
  usernameExists=""
  usernameValidationError="Username required and must be 3-72 characters in length."
>
  <p
    className="ValidateUserName__validator error"
    id="username-validation"
  >
    Username required and must be 3-72 characters in length.
    <br />
  </p>
</ValidateUsername>
`;
